pkg/nfd-topology-updater/nfd-topology-updater.go

```diff
@@ -88,17 +88,24 @@ type nfdTopologyUpdater struct {
 	apihelper           apihelper.APIHelpers
 	resourcemonitorArgs resourcemonitor.Args
 	stop                chan struct{} // channel for signaling stop
+	eventSource         <-chan kubeletnotifier.Info
 	configFilePath      string
 	config              *NFDConfig
 }
 
 // NewTopologyUpdater creates a new NfdTopologyUpdater instance.
-func NewTopologyUpdater(args Args, resourcemonitorArgs resourcemonitor.Args, policy, scope string) NfdTopologyUpdater {
+func NewTopologyUpdater(args Args, resourcemonitorArgs resourcemonitor.Args, policy, scope string, eventSource chan kubeletnotifier.Info) NfdTopologyUpdater {
+	ntf, err := kubeletnotifier.New(resourcemonitorArgs.SleepInterval, eventSource, args.KubeletDirPath)
+	if err != nil {
+		klog.Exit(err)
```

Comment:
this is a bit brutal, I'd just bubble up the error and let the caller decide if abort everything (likely) or not

Reply:
Yeah, change `NewTopologyUpdater()` to return an error